diff --git a/assets/app.js b/assets/app.js
index 0625944..edc3b00 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -11,6 +11,11 @@ import './styles/app.css';
 // start the Stimulus application
 import './bootstrap';
 
+window.addEventListener('DOMContentLoaded', (event) => {
+    console.log('page is ready!');
+});
+console.log('script is done!');
+
 const flockEl = document.getElementById('flock');
 document.querySelector('.js-grow-flock').addEventListener('click', (event) => {
     event.preventDefault();
