diff --git a/assets/app.js b/assets/app.js
index b4f6aad..bb0a6aa 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -10,13 +10,3 @@ import './styles/app.css';
 
 // start the Stimulus application
 import './bootstrap';
-
-const syncButton = (linkElement) => {
-    document.querySelector('.js-custom-animal-target').innerHTML = linkElement.value || '☹️';
-}
-document.querySelector('.js-custom-animal').addEventListener('keyup', (event) => {
-    syncButton(event.currentTarget);
-});
-document.querySelector('.js-custom-animal').addEventListener('change', (event) => {
-    syncButton(event.currentTarget);
-});
diff --git a/assets/controllers/sync-button-controller.js b/assets/controllers/sync-button-controller.js
new file mode 100644
index 0000000..d5b5606
--- /dev/null
+++ b/assets/controllers/sync-button-controller.js
@@ -0,0 +1,9 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+    static targets = ['button'];
+
+    sync(event) {
+        this.buttonTarget.innerHTML = event.currentTarget.value || '☹️';
+    }
+}
diff --git a/templates/default/index.html.twig b/templates/default/index.html.twig
index d99dff7..c7251a6 100644
--- a/templates/default/index.html.twig
+++ b/templates/default/index.html.twig
@@ -24,15 +24,20 @@
             {% endif %}
         {% endfor %}
 
+        <div {{ stimulus_controller('sync-button') }}>
         {{ form_start(form) }}
             or custom
             <input
                 name="{{ field_name(form.animal) }}"
                 value="{{ field_value(form.animal) }}"
-                class="js-custom-animal {{ field_errors(form.animal)|length > 0 ? 'has-error' }}"
+                class="{{ field_errors(form.animal)|length > 0 ? 'has-error' }}"
+                {{ stimulus_action({
+                    'sync-button': { 'keyup': 'sync', 'change': 'sync' }
+                }) }}
             />
 
-            <button type="submit" class="js-custom-animal-target" style="font-size: 1em;" formnovalidate>🦖</button>
+            <button type="submit" {{ stimulus_target('sync-button', 'button')}} style="font-size: 1em;" formnovalidate>🦖</button>
         {{ form_end(form) }}
+        </div>
     </div>
 {% endblock %}
