diff --git a/assets/controllers/react-footer-controller.js b/assets/controllers/react-footer-controller.js
index 7889753..6f29abb 100644
--- a/assets/controllers/react-footer-controller.js
+++ b/assets/controllers/react-footer-controller.js
@@ -4,6 +4,7 @@ import { render } from 'react-dom';
 import { PrintMessage } from '../components/PrintMessage';
 import React from 'react';
 
+/* stimulusFetch: 'lazy' */
 export default class extends Controller {
     static values = {
         message: String
