diff --git a/assets/app.js b/assets/app.js
index bb0a6aa..0625944 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -10,3 +10,20 @@ import './styles/app.css';
 
 // start the Stimulus application
 import './bootstrap';
+
+const flockEl = document.getElementById('flock');
+document.querySelector('.js-grow-flock').addEventListener('click', (event) => {
+    event.preventDefault();
+
+    flockEl.innerHTML = flockEl.innerHTML + flockEl.dataset.animal;
+});
+
+const syncButton = (linkElement) => {
+    document.querySelector('.js-custom-animal-target').innerHTML = linkElement.value || '☹️';
+}
+document.querySelector('.js-custom-animal').addEventListener('keyup', (event) => {
+    syncButton(event.currentTarget);
+});
+document.querySelector('.js-custom-animal').addEventListener('change', (event) => {
+    syncButton(event.currentTarget);
+});
diff --git a/templates/default/index.html.twig b/templates/default/index.html.twig
index e24ab12..a7a71a6 100644
--- a/templates/default/index.html.twig
+++ b/templates/default/index.html.twig
@@ -5,7 +5,7 @@
         <h1>Grow your Flock of {{ chosenAnimal }}</h1>
 
         <div>
-            <div class="flock" id="flock">
+            <div class="flock" id="flock" data-animal="{{ chosenAnimal }}">
                 {{ chosenAnimal }}
             </div>
             <button class="js-grow-flock">Grow my flock!</button>
@@ -27,10 +27,10 @@
             <input
                 name="{{ field_name(form.animal) }}"
                 value="{{ field_value(form.animal) }}"
-                class="{{ field_errors(form.animal)|length > 0 ? 'has-error' }}"
+                class="js-custom-animal {{ field_errors(form.animal)|length > 0 ? 'has-error' }}"
             />
 
-            <button type="submit" style="font-size: 1em;" formnovalidate>🦖</button>
+            <button type="submit" class="js-custom-animal-target" style="font-size: 1em;" formnovalidate>🦖</button>
         {{ form_end(form) }}
     </div>
 {% endblock %}
